python2/httplib2/__init__.py

```diff
@@ -1740,9 +1756,106 @@ def clear_credentials(self):
         self.credentials.clear()
         self.authorizations = []
 
+    def add_cookie(self, name, value, max_age=None, expires=None):
+        """Set new cookie or update the existing cookie value.
+
+        max_age:
+            Number of seconds until the cookie expires
+        expires:
+            The maximum lifetime of the cookie as an HTTP-date timestamp.
+            Format:
+                <day-name>, <day> <month> <year> <hour>:<minute>:<second> GMT
+            Example: Wed, 21 Oct 2015 07:28:00 GMT
+
+        max_age has higher priority than expires.
+        """
+        if max_age:
+            expire_time = time.time() + int(max_age)
+        elif expires:
+            expire_time = calendar.timegm(email.Utils.parsedate_tz(expires))
+        else:
+            expire_time = None
+
+        self.cookies[name] = (value, expire_time)
```

Comment:
You'd need to store another `secure` flag with a cookie, so namedtuple is probably best storage for individual cookie item.